Three/include/CGAL/Three/Scene_item.h

```diff
@@ -305,7 +305,7 @@ public Q_SLOTS:
   //!Setter for the name of the item.
   virtual void setName(QString n) { name_ = n; }
   //!Setter for the visibility of the item.
-  virtual void setVisible(bool b) { visible_ = b; }
+  virtual void setVisible(bool b) { visible_ = b; Q_EMIT itemChanged();}
   //!Setter for the rendering mode of the item.
```

Comment:
This seems very suspicious!

Generally the `Scene_item::setVisible()` is called by the scene. In that case there is no need to call `itemChanged()` because the scene already know.

In the current case, `setVisible` is called the main window, which calls `itemChanged()` as well, so that the scene know.

Probably the `Scene_group_item` should deal with that. But anyway, I disagree with the call `Scene_item::itemChanged()` from `Scene_item::setVisible`, unless we change the semantic of `itemChanged` (but that is another topic).